nautobot/extras/models/datasources.py

```diff
@@ -81,13 +82,24 @@ def __str__(self):
     def clean(self):
         super().clean()
 
-        if self.slug != "":
-            check_if_key_is_graphql_safe(self.__class__.__name__, self.slug, "slug")
+        # Autogenerate slug now, rather than in pre_save(), if not set already, as we need to check it below.
+        if self.slug == "":
+            self._meta.get_field("slug").create_slug(self, add=(not self.present_in_database))
+
+        if self.present_in_database and self.slug != self.__initial_slug:
+            raise ValidationError(
+                f"Slug cannot be changed once set. Current slug is {self.__initial_slug}, "
+                f"requested slug is {self.slug}"
+            )
 
-            if self.present_in_database and self.slug != self.__initial_slug:
+        if not self.present_in_database:
+            check_if_key_is_graphql_safe(self.__class__.__name__, self.slug, "slug")
+            # Check on create whether the proposed slug conflicts with a module name already in the Python environment.
+            # Because we add GIT_ROOT to the end of sys.path, trying to import this repository will instead
+            # import the earlier-found Python module in its place, which would be undesirable.
+            if find_spec(self.slug) is not None:
                 raise ValidationError(
-                    f"Slug cannot be changed once set. Current slug is {self.__initial_slug}, "
-                    f"requested slug is {self.slug}"
+                    f'Please choose a different slug, as "{self.slug}" is an existing Python package or module.'
```

Comment:
Don't know what would be a better message here but since we aren't inspecting all potential packages or modules I'm wondering if a suggestion that the resulting slug is a protected name would be better.

Reply:
Would changing `existing` to `installed` be an improvement?